cmd/deploy.go

```diff
@@ -562,20 +538,39 @@ func parseImageDigest(imageSplit []string, config deployConfig, cmd *cobra.Comma
 // checkNamespaceDeploy checks current namespace against func.yaml and warns if its different
 // or sets namespace to be written in func.yaml if its the first deployment
 func checkNamespaceDeploy(funcNamespace string, confNamespace string) (string, error) {
-	currNamespace, err := k8s.GetNamespace("")
-	if err != nil {
-		return funcNamespace, err
+	var namespace string
+	var err error
+
+	// Choose target namespace
+	if confNamespace != "" {
+		namespace = confNamespace // --namespace takes precidence
+	} else if funcNamespace != "" {
+		namespace = funcNamespace // value from previous deployment (func.yaml) 2nd priority
+	} else {
+		// Try to derive a default from the current k8s context, if any.
+		namespace, err = k8s.GetNamespace("")
+		if err != nil {
+			return "", err
+		}
```

Comment:
We may want to treat this as nonfatal (perhaps with a warning printed to stderr?)